lib/StringClient.js

```diff
@@ -234,7 +237,8 @@ StringClient.prototype.parse = function parse(req, callback) {
 
             res.on('data', function onData(chunk) {
                 if (hash) {
-                    hash.update(chunk);
+                    hash.update(chunk, 'utf8');
```

Comment:
A comment here explaining why we want to bother calculating both hashes would be helpful for future maint.
Likewise a quick comment in the test files explaining *why* "it should validate content-md5 with XXX encoding" would be helpful.